### ISSUE TITLE
AWS API breaking change

### ISSUE THREAD

**Issue by 80kk** [cloudposse/terraform-aws-elasticache-redis#197]
### Describe the Bug

in `aws_elasticache_replication_group` resource there is a deprecated [availability_zones](https://registry.terraform.io/providers/hashicorp/aws/4.67.0/docs/resources/elasticache_replication_group#availability_zones) argument.  Apparently its been now removed from the AWS API completely and it is causing an error:
```
│ Error: Unsupported argument
│ 
│   on .terraform/modules/[REDACTED].redis/main.tf line 125, in resource "aws_elasticache_replication_group" "default":
│  125:   availability_zones         = length(var.availability_zones) == 0 ? null : [for n in range(0, var.cluster_size) : element(var.availability_zones, n)]
│ 
│ An argument named "availability_zones" is not expected here.
```
Downgrading aws provider to older one won't help.

### Expected Behavior

It should be replaced with `preferred_cache_cluster_azs` as mentioned in Terraform docs. It looks like its been already proposed: https://github.com/cloudposse/terraform-aws-elasticache-redis/pull/174

### Steps to Reproduce

Run `terraform plan`

**Comment by odie5533** [comment]
Related to #173

**Comment by mosheven** [comment]
Bug resolved on releases 0.51.1
https://github.com/cloudposse/terraform-aws-elasticache-redis/releases/tag/0.51.1
